tests/mc_comm/test_mc_comm.py

```diff
@@ -1134,3 +1148,194 @@ def test_McCommunicationProcess__raises_error_if_status_beacon_not_received_in_a
     )
     with pytest.raises(SerialCommStatusBeaconTimeoutError):
         invoke_process_run_and_check_errors(mc_process)
+
+
+def test_McCommunicationProcess__processes_reboot_command(
+    four_board_mc_comm_process_no_handshake, mantarray_mc_simulator, mocker
+):
+    mc_process = four_board_mc_comm_process_no_handshake["mc_process"]
+    board_queues = four_board_mc_comm_process_no_handshake["board_queues"]
+    simulator = mantarray_mc_simulator["simulator"]
+    input_queue = board_queues[0][0]
+    output_queue = board_queues[0][1]
+
+    mocker.patch.object(
+        mc_simulator,
+        "_get_secs_since_reboot_command",
+        autospec=True,
+        side_effect=[MAX_MC_REBOOT_DURATION_SECONDS / 2],
+    )
+    set_connection_and_register_simulator(mc_process, mantarray_mc_simulator)
+
+    expected_response = {
+        "communication_type": "to_instrument",
+        "command": "reboot",
+    }
+    put_object_into_queue_and_raise_error_if_eventually_still_empty(
+        copy.deepcopy(expected_response), input_queue
+    )
+    # run mc_process one iteration to send reboot command
+    invoke_process_run_and_check_errors(mc_process)
+    # run simulator to start reboot and mc_process to confirm reboot started
+    invoke_process_run_and_check_errors(simulator)
+    invoke_process_run_and_check_errors(mc_process)
+    confirm_queue_is_eventually_of_size(output_queue, 1)
+    command_response = output_queue.get_nowait()
```

Comment:
`get_nowait` in the test suite sometimes causes problems right? don't we usually use `get(timeout)`